### PR TITLE
fix(bm): patch of security groups on bare metal server nics

### LINKED ISSUE #4479: ibm_is_bare_metal_server: change to security groups does not trigger change
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform IBM Provider Version

```
Terraform v1.3.7
on darwin_amd64
+ provider registry.terraform.io/hashicorp/local v2.4.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.4.3
+ provider registry.terraform.io/hashicorp/tls v4.0.4
+ provider registry.terraform.io/ibm-cloud/ibm v1.52.0
```

### Affected Resource(s)

* ibm_is_bare_metal_server

### Debug Output

```
  # ibm_is_bare_metal_server.bm_vpc_host1 will be updated in-place
  ~ resource "ibm_is_bare_metal_server" "bm_vpc_host1" {
        id                 = "0737-zzzz-zzzz-zzzz-zzzz-zzzzzz"
        name               = "bm1"
        tags               = []
        # (19 unchanged attributes hidden)

      ~ primary_network_interface {
            id                        = "0737-zzzz-zzzz-zzzz-zzzz-zzzzzz"
            name                      = "garment-crying-relic-month"
          ~ security_groups           = [
              - "r006-zzzz-zzzz-zzzz-zzzz-zzzzzz",
              - "r006-zzzz-zzzz-zzzz-zzzz-zzzzzz",
                # (2 unchanged elements hidden)
            ]
            # (7 unchanged attributes hidden)

            # (1 unchanged block hidden)
        }

        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

### Expected Behavior

ibm_is_bare_metal_server: change to security groups

### Actual Behavior

ibm_is_bare_metal_server: change to security groups does not trigger change

### Steps to Reproduce

Any execution of Terraform Resource `ibm_is_bare_metal_server` where Security Groups are updated

### PR DESCRIPTION
**Pull request by ujjwal-ibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #4479

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: ibm/service/vpc/resource_ibm_is_bare_metal_server.go [hunk 1]
```diff
@@ -2388,32 +2389,79 @@ func bareMetalServerUpdate(context context.Context, d *schema.ResourceData, meta
...
 			}
 		}
 		if d.HasChange("primary_network_interface.0.enable_infrastructure_nat") {
 			if enableNatOk, ok := d.GetOk("primary_network_interface.0.enable_infrastructure_nat"); ok {
 				enableNat := enableNatOk.(bool)
 				bmsNicPatchModel.EnableInfrastructureNat = &enableNat
+				nicflag = true
+			}
+		}
+		if d.HasChange("primary_network_interface.0.security_groups") && !d.IsNewResource() {
```
Comment (hkantare) [inline ibm/service/vpc/resource_ibm_is_bare_metal_server.go:2402]: This change doesnt require to set  				nicflag = true

**Comment by hkantare** [comment]
Can you add test case results

**Comment by ujjwal-ibm** [comment]
```
=== RUN   TestAccIBMISBareMetalServer_sg_update
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
--- PASS: TestAccIBMISBareMetalServer_sg_update (1092.94s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     1094.963s
```

File: ibm/service/vpc/resource_ibm_is_bare_metal_server.go (hunk 1 above)
Comment (ujjwal-ibm) [reply ibm/service/vpc/resource_ibm_is_bare_metal_server.go:2402]: No, adding of sg groups is taken care inside this block itself.

**Review by hkantare — APPROVED** [review]

**Pull request merged**